File: ruby-pr/53/ruby/code-quality-readability-maintainability/data_processor.rb
class DataProcessor
  def process_tasks(tasks)
    # {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
    (0...tasks.length).each do |i|
      task = tasks[i]
      status = task[:status]
      priority = task[:priority]
      type = task[:type]
      
      if status == 'pending'
        if priority == 1
          if type == 'urgent'
            puts "Processing urgent high priority task"
            task[:status] = 'processing'
          else
            puts "Processing high priority task"
            task[:status] = 'processing'
          end
        elsif priority == 2
          puts "Processing medium priority task"
          task[:status] = 'processing'
        else
          puts "Processing low priority task"
          task[:status] = 'processing'
        end
      end
    end
    # {/fact}
  end
end

File: php-pr/51/php/code-quality-readability-maintainability/DataProcessor.php
<?php

class DataProcessor
{
    public function processTasks($tasks)
    {
        // {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
        for ($i = 0; $i < count($tasks); $i++) {
            $task = $tasks[$i];
            $status = $task['status'];
            $priority = $task['priority'];
            $type = $task['type'];
            
            if ($status == 'pending') {
                if ($priority == 1) {
                    if ($type == 'urgent') {
                        echo "Processing urgent high priority task\n";
                        $task['status'] = 'processing';
                    } else {
                        echo "Processing high priority task\n";
                        $task['status'] = 'processing';
                    }
                } else if ($priority == 2) {
                    echo "Processing medium priority task\n";
                    $task['status'] = 'processing';
                } else {
                    echo "Processing low priority task\n";
                    $task['status'] = 'processing';
                }
            }
        }
        // {/fact}
    }
}